org.eclipse.xtext.common.types.ui/src/org/eclipse/xtext/common/types/xtext/ui/JdtTypesProposalProvider.java

```diff
@@ -243,13 +241,9 @@ protected IJavaSearchScope createSearchScope(IJavaProject project, JvmType super
 			return new IntersectingJavaSearchScope(); // empty intersection
 		}
 		try {
-			// Method is available in JDT 3.6 or better
-			// more than twice as fast as the older alternative
-			Method method = SearchEngine.class.getMethod("createStrictHierarchyScope", IJavaProject.class, IType.class, Boolean.TYPE, Boolean.TYPE, WorkingCopyOwner.class);
-			method.setAccessible(true);
-			IJavaSearchScope result = (IJavaSearchScope) method.invoke(null, project, type, Boolean.TRUE, Boolean.TRUE, null);
+			IJavaSearchScope result = SearchEngine.createStrictHierarchyScope(project, type, Boolean.TRUE, Boolean.TRUE, null);
 			return result;
-		} catch (Exception e) {
+		} catch (JavaModelException e) {
```

Comment:
I think most of the stuff in the catch block was for older jdt versions. Shouldn't this be removed?

Reply:
i dont see how this is connected to reflection.

Reply:
The caught exception was for the missing method. Now we catch JavaModelException, but that's also declared on the method signature and likely to be thrown again by the code in the catch block.

Reply:
i am not sure. the old could fail with java model exception or reflective exception

Reply:
True. Point taken.

Reply:
and JavaModelException extends Exception

Reply:
so you would simply remove the catch?

Reply:
Yes, that's what I would have done. But it could have been the wrong thing to do in this case. So  you convinced me that the fallback could still be valuable.

Reply:
;+1: